dali/kernels/imgproc/flip_test.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_FLIP_TEST_H_
+#define DALI_KERNELS_IMGPROC_FLIP_TEST_H_
+
+namespace dali {
+namespace kernels {
+
+template <typename T>
+bool is_flipped(const T* lhs, const T* rhs, size_t layers, size_t height, size_t width,
+                size_t channels, bool flip_z, bool flip_y, bool flip_x) {
```

Comment:
Why not use homogeneous coordinates and transformation matrix to do this? That way it would be extremely clear, that it's the right test
I mean transformation matrices like:
```
Horizontal = I(3)
Horizontal[0][0] = -1
Vertical = I(3)
Vertical[1][1] = -1
```
where `I(3)` is 3x3 identity matrix